Braden Walker

bradencodes opened the pull request:
## Trello Set Up:

* [x] Create a Trello account
* [x] Create a new board called "Lambda Notes - {Your Name}"
* [x] Create lists titled `backlog`,`To Do`, `In Progress`, and `Done`
* [x] Fill in the `To Do` list with the MVP features listed below
* [x] Fill in the `backlog` list with all the extra features listed below
* [x] Share your board with the project manager that has been assigned to you. If you have not been assigned yet, reach out to your lead TA for guidance
* [x] Add your Trello URL to your project's README.md file. Commit the change, push it to your repository & submit a pull request

## MVP Features:

* [x] Display a list of notes
* [x] Create a note with a title and content
* [x] View an existing note
* [x] Edit an existing note
* [x] Delete an existing note

Commit:
added trello url to README

Commit:
initialized project, added libraries and dummy data

Commit:
connected google fonts Roboto and Raleway

Commit:
created Sidebar with reactstrap

Commit:
finished pixel perfect sidebar

Commit:
finished pixel perfect listview

Commit:
finished pixel perfect ListView, CreateView, and NoteView

Commit:
refactored css to be non-conflicting

Commit:
linked cards to noteView

Commit:
refactored CreateNote.css to be non-conflicting

Commit:
refactored sidebar Links to match button boundary

Commit:
refactored createNote Link to fit button boundary

Commit:
refactored ListView so links fit card boundary

Commit:
set max-width on app to keep pixel perfection

Commit:
reconfigured components to be pixel perfect

Commit:
finished pixel perfect delete modal

Commit:
finished pixel perfect EditNote component

MVP is done!

Commit:
made everything responsive

Commit:
listView and noteView now functional with JSON data

enitchals commented on lambdanotes/src/App.js:
```diff
@@ -0,0 +1,46 @@
+    super();
+    this.state = {
+      notesList: [],
+      currentID: 0
```
I assume you're using this to keep track of which single note you're looking at? Very nice :+1:

Commit:
refactored listView and noteView to connect with server

Commit:
createNote now creates a new note on the server

Commit:
added delete functionality with server

delete button is exhibiting some weird behavior. Sometimes the note will be deleted. Other times the note will stay and I have to try and delete it over and over until it works. The note always gets deleted, but it takes anywhere from 1 to ~20 tries before working

Commit:
fixed .delete issue

I put the window.location.reload inside of .then so it would run after the promise was fulfilled

Commit:
added full functionality to editNote component with server

Commit:
turned p into span and scrolling listView notes

Commit:
changed scrolling behavior and routing from editNote

edit and delete in noteView dont scroll with the note. After editing a note, it goes straight to that notes view

Commit:
added marked and react-marked packages for markdown

Commit:
added markdown support for noteView

Commit:
added markdown support to listView

Commit:
images now fit boundaries

Commit:
made modal corners square

Commit:
replaced window reload with updating App state

this way the window will not flash when it needs to update

Commit:
added sorting options in listView

Commit:
added search functionality

Commit:
styled the search bar

Commit:
fixed listView styling

Commit:
connected to lamb-danotes on heroku

Commit:
added register and login components with proper authorization

Commit:
addout logout button and revised some css

Commit:
placed react-app into the root

bradencodes committed:
add environment variable for API URL

bradencodes committed:
app is working again!

bradencodes committed:
change name of app on sidebar

bradencodes committed:
take out API endpoint comments

bradencodes committed:
fix pointer when there are no notes
